mapfishapp/src/main/webapp/app/js/GEOR_Querier.js

```diff
@@ -0,0 +1,322 @@
+ *   Styler/widgets/form/SpatialComboBox.js, which is
+ *   included by Styler/widgets/SpatialFilterPanel.js, requires it
+ * - -AT-include OpenLayers/Filter/Logical.js is required because
+ *   Styler/widgets/FilterBuilder.js uses OpenLayers.Filter.Logical
+ */
+
+/*
+ * @requires OpenLayers/Filter/Comparison.js
+ * @requires OpenLayers/Filter/Spatial.js
+ * @include OpenLayers/Filter/Logical.js
+ * @include OpenLayers/Format/Filter.js
+ * @include OpenLayers/Format/XML.js
+ // see https://github.com/georchestra/georchestra/issues/482 :
+ * @include OpenLayers/Format/WKT.js
+ * @include OpenLayers/Control/ModifyFeature.js
+ * @include OpenLayers/Control/DrawFeature.js
+ * @include OpenLayers/Handler/Point.js
+ * @include OpenLayers/Handler/Path.js
+ * @include OpenLayers/Handler/Polygon.js
+ * @include GeoExt/data/LayerRecord.js
+ * @include GeoExt/data/AttributeStore.js
+ * @include Styler/widgets/FilterBuilder.js
+ * @include GEOR_waiter.js
+ * @include GEOR_ows.js
+ * @include GEOR_util.js
+ */
+
+/* globals Ext, GeoExt, OpenLayers, GEOR */
+
+Ext.namespace("GEOR");
+
+
+GEOR.Querier = Ext.extend(Ext.Window, {
+
+    layout: "fit",
+    border: false,
+    closeAction: "close",
+    // the local Styler.FilterBuilder instance
+    filterbuilder: null,
+    cookieProvider: null,
+    pseudoRecord: null,
+
+    /**
+     * Property: geometryName
+     * {String} The geometry column name
+     */
+    geometryName: null,
+
+    /**
+     * Property: record
+     * {Ext.data.Record} the WMS or WFS layer record
+     */
+    record: null,
+
+    /**
+     * Property: map
+     * {OpenLayers.Map} The map instance.
+     */
+    map: null,
+
+    /**
+     * Property: layer
+     * {OpenLayers.Layer.Vector} The layer on which geometries can be drawn
+     */
+    layer: null,
+
+    /**
+     * Property: attributeStore
+     * {GeoExt.data.AttributeStore} 
+     */
+    attributeStore: null,
+    
+    /**
+     * Property: layerFields
+     * {Array} an array of fields for the current layer
+     * It is extracted from the WFS DescribeFeatureType operation
+     */  
+    layerFields: null,
+
+    /*
+     * Method: initComponent.
+     * Overridden constructor. Set up widgets and lay them out
+     */
+    initComponent: function() {
+        var r = this.record,
+            type = r.get("type"),
+            isWFS = type === "WFS",
+            layer = r.get("layer"), 
+            name = r.get("title") || layer.name || "";
+
+        this.title = OpenLayers.i18n("Request on NAME", {
+            "NAME": name
+        });
+
+        this.pseudoRecord = {
+            typeName: isWFS ? 
+                r.get("WFS_typeName") : r.get("name"),
+            owsURL: isWFS ? 
+                layer.protocol.url : r.get("WFS_URL")
+        };
+
+        GEOR.waiter.show();
+        // get layer model through WFS DescribeFeatureType:
+        this.attributeStore = GEOR.ows.WFSDescribeFeatureType(this.pseudoRecord, {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [A function with a name starting with an uppercase letter should only be used as a constructor.](https://www.codacy.com/app/georchestra/georchestra/file/2337895343/issues/source?bid=3200181&fileBranchId=3209168#l125)